antarest/study/business/st_storage_manager.py

```diff
@@ -445,29 +445,30 @@ def update_storage(
         values = new_config.dict(by_alias=False)
         return StorageOutput(**values)
 
-    def delete_storage(
+    def delete_storages(
         self,
         study: Study,
         area_id: str,
-        storage_id: str,
+        form: List[str],
     ) -> None:
         """
-        Delete a short-term storage configuration form the given study and area_id.
+        Delete  short-term storages configuration form the given study and area_id.
 
         Args:
             study: The study object.
             area_id: The area ID of the short-term storage.
-            storage_id: The ID of the short-term storage to remove.
+            form: list ID of  short-term storages to remove.
         """
-        command = RemoveSTStorage(
-            area_id=area_id,
-            storage_id=storage_id,
-            command_context=self.storage_service.variant_study_service.command_factory.command_context,
-        )
-        file_study = self.storage_service.get_storage(study).get_raw(study)
-        execute_or_add_commands(
-            study, file_study, [command], self.storage_service
-        )
+        for storage_id in form:
+            command = RemoveSTStorage(
+                area_id=area_id,
+                storage_id=storage_id,
+                command_context=self.storage_service.variant_study_service.command_factory.command_context,
```

Comment:
You can optimize performance by extracting the context into a `command_context` local variable outside the loop, thus avoiding the need to calculate it on each iteration.